Content/Items/Misc/Weapons.Soilgun.cs

```diff
@@ -0,0 +1,1488 @@
+                }
+            }
+            if (CurrentCharge < 2)
+                return;
+            int DustFrequency = (int)(15 - (Utils.Clamp(CurrentCharge / 5, 0, 12)));
+            if (Main.rand.NextBool(Utils.Clamp(DustFrequency, 1, 15)))
+            {
+                Dust dust = Dust.NewDustDirect(barrelPos, 2, 8, DustID.Dirt, 0f, 0f);
+                dust.scale = Main.rand.NextFloat(0.8f, 1.2f);
+                dust.noGravity = false;
+                if (Main.rand.NextBool(5))
+                    Dust.NewDustDirect(barrelPos, 2, 8, ModContent.DustType<Dusts.Sand>(), 0, 0, 125, default, 0.5f);
+            }
+        }
+
+        public void ShootThings(Vector2 position)
+        {
+            if (Main.myPlayer != Projectile.owner)
+                return;
+            Item heldItem = owner.HeldItem;
+            int damage = Projectile.damage;
+            float shootSpeed = heldItem.shootSpeed;
+            float knockBack = owner.GetWeaponKnockback(heldItem, heldItem.knockBack);
+            Vector2 shootVelocity = Utils.SafeNormalize(Projectile.velocity, Vector2.UnitY) * shootSpeed;
+            for (int i = 0; i < 4 + Main.rand.Next(3); i++)
+            {
+                Projectile.NewProjectile(Projectile.GetSource_FromThis(), position, (shootVelocity.RotatedByRandom(MathHelper.ToRadians(18))) * Main.rand.NextFloat(0.9f, 1.1f), ModContent.ProjectileType<SoilGunSoilProjs>(), damage, knockBack, owner.whoAmI, SoilType);
+            }
+            for (int i = 0; i < 15; i++)
+            {
+                Vector2 dustVelocity = shootVelocity.RotatedByRandom(MathHelper.ToRadians(8)) * Main.rand.NextFloat(0.25f, 0.45f);
+                Dust dust = Dust.NewDustDirect(position, 2, 8, DustID.Dirt, dustVelocity.X, dustVelocity.Y);
+                dust.scale = Main.rand.NextFloat(1.1f, 1.55f);
+                dust.noGravity = true;
+            }
+            for (int i = 0; i < 7; i++)
+            {
+                Vector2 sandVelocity = (shootVelocity * Main.rand.NextFloat(0.12f, 0.20f)).RotatedByRandom(MathHelper.ToRadians(9f));
+                Dust.NewDustDirect(position, 2, 8, ModContent.DustType<Dusts.Sand>(), sandVelocity.X, sandVelocity.Y, 140, default, 0.5f);
+
+                Dust.NewDustDirect(position, 2, 8, ModContent.DustType<SandNoGravity>(), sandVelocity.X, sandVelocity.Y, 135, default, 0.65f);
+
+                Dust.NewDustDirect(position, 2, 8, ModContent.DustType<SandNoGravity>(), sandVelocity.X, sandVelocity.Y, 135, default, 0.55f);
+            }
+            CameraSystem.Shake += 2;
+            owner.reuseDelay = 15;
+            Item item = owner.GetModPlayer<SoilgunPlayer>().SoilItem;
+            item.stack--;
+            if (item.stack <= 1)
+            {
+                item.TurnToAir();
+            }
+            //need a better sfx
+            SoundEngine.PlaySound(SoundID.Item61, Projectile.position);
+        }
+    }
+
+    class SandNoGravity : ModDust
```

Comment:
This should be moved to its own file in the dust directory or atleast a file seperate to this. Consider the folder where the projectiles will likely need to be moved to cut down on this files size.